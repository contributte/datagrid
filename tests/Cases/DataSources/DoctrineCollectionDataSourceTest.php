<?php

namespace Ublaboo\DataGrid\Tests\Cases\DataSources;
use Doctrine\Common\Collections\ArrayCollection;
use Ublaboo;

require __DIR__ . '/BaseDataSourceTest.phpt';

final class DoctrineCollectionDataSourceTest extends BaseDataSourceTest
{
	public function setUp()
	{

		$this->ds = new Ublaboo\DataGrid\DataSource\DoctrineCollectionDataSource( new ArrayCollection($this->data),'id');
		$factory = new Ublaboo\DataGrid\Tests\Files\XTestingDataGridFactory;
		$this->grid = $factory->createXTestingDataGrid();
	}
}
$test_case = new DoctrineCollectionDataSourceTest();
$test_case->run();
