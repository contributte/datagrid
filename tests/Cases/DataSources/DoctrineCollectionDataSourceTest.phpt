<?php declare(strict_types = 1);

namespace Contributte\Datagrid\Tests\Cases\DataSources;

use Contributte\Datagrid\DataSource\DoctrineCollectionDataSource;
use Contributte\Datagrid\Filter\FilterText;
use Contributte\Datagrid\Tests\Files\TestingDatagridFactory;
use Doctrine\Common\Collections\ArrayCollection;
use Tester\Assert;

require __DIR__ . '/BaseDataSourceTest.phpt';

final class DoctrineCollectionDataSourceTest extends BaseDataSourceTest
{

	public function setUp(): void
	{
		$this->ds = new DoctrineCollectionDataSource(new ArrayCollection($this->data), 'id');
		$factory = new TestingDatagridFactory();
		$this->grid = $factory->createTestingDatagrid();
	}

	public function testFilterMultipleColumns(): void
	{
		$filter = new FilterText($this->grid, 'a', 'b', ['name', 'address']);
		$filter->setValue('lu');
		$this->ds->filter([$filter]);

		Assert::equal([
			$this->data[0],
			$this->data[4],
		], $this->getActualResultAsArray());
	}

}

(new DoctrineCollectionDataSourceTest())->run();
