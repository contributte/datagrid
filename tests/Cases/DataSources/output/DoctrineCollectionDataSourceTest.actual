[
	[
		'id' => 1,
		'name' => 'John Doe',
		'age' => 30,
		'address' => 'Blue Village 1',
	],
	[
		'id' => 30,
		'name' => 'Jackie Blue',
		'age' => 80,
		'address' => 'Prague 678',
	],
]
