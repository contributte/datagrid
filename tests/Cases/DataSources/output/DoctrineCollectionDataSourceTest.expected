[
	[
		'id' => 1,
		'name' => 'John Doe',
		'age' => 30,
		'address' => 'Blue Village 1',
	],
	[
		'id' => 8,
		'name' => 'Jude Law',
		'age' => 8,
		'address' => 'Lubababa 5',
	],
	[
		'id' => 30,
		'name' => 'Jackie Blue',
		'age' => 80,
		'address' => 'Prague 678',
	],
]
